### PR TITLE
fix: slider track draggale range height/width error

### LINKED ISSUE #907: Unpredictable behaviour of a draggable track slider
### Reproduction link

https://codesandbox.io/s/draggable-slider-izgbr4?file=/index.js

### Steps to reproduce

Try to click and drag the selected interval a couple of times. 

### What is expected?

There should be some fixed maximum distance (could be a percentage of the selected interval or an absolute value in pixels) which controls whether the endpoint should be moved or the entire interval should be dragged.

### What is actually happening?

You will notice that sometimes it's draggable and sometimes one of the end positions moves onto where you clicked.

### PR DESCRIPTION
**Pull request by BoyYangzai**
- fix https://github.com/ant-design/ant-design/issues/42628
- fix #907

### CODE REVIEW

File: src/Tracks/Track.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ export default function Track(props: TrackProps) {
   const { direction, min, max, disabled, range } = React.useContext(SliderContext);
 
   const trackPrefixCls = `${prefixCls}-track`;
+  const draggableTrackPrefixCls = onStartMove ? `${prefixCls}-track-draggable` : null;
```
Comment (yoyo837) [inline src/Tracks/Track.tsx:21]: 把三元判断直接改为在下面的classNames 写判断吧？

Comment (yoyo837) [reply src/Tracks/Track.tsx:21]: 单看下面还以为一直有

File: assets/index.less [hunk 2]
```diff
@@ -175,6 +184,15 @@
       left: 5px;
       width: 4px;
     }
+    
+    &-track-draggable {
+      z-index: 1;
```
Comment (yoyo837) [inline assets/index.less:189]: 两坨样式看上去是一样的，能写成一处并达到相同效果吗？可以减少后续因为漏改一处的风险

File: src/Tracks/Track.tsx (hunk 1 above)
Comment (BoyYangzai) [reply src/Tracks/Track.tsx:21]: 会不会导致 classNames 这里过于臃肿了

File: assets/index.less (hunk 2 above)
Comment (BoyYangzai) [reply assets/index.less:189]: 嗯 antd这一层是 js 写 我用了三元让它写在了一处
这个地方是less 要根据他 上一层的一个叫 vertical 的类 下的 &-track-draggable 让它在垂直状态时获得全新的样式 
@zombieJ 这里有办法写成一出吗

File: src/Tracks/Track.tsx [hunk 3]
```diff
@@ -55,7 +56,11 @@ export default function Track(props: TrackProps) {
 
   return (
     <div
-      className={classNames(trackPrefixCls, range && `${trackPrefixCls}-${index + 1}`)}
+      className={classNames(
+        trackPrefixCls,
+        range && `${trackPrefixCls}-${index + 1}`,
+        draggableTrackPrefixCls,
+      )}
```
Comment (Wxh16144) [inline src/Tracks/Track.tsx:63]:
```suggestion
      className={classNames(
        trackPrefixCls,
        {
          [`${trackPrefixCls}-${index + 1}`]: range,
          [`${prefixCls}-track-draggable`]: onStartMove
        }
      )}
```

**Review by MadCcc — APPROVED** [review]

**Comment by miloszmaki** [comment]
@BoyYangzai thanks for the fix, dragging works well now :+1: 
however there is some subtle rendering issue now, the track is being rendered over the handles, see https://codesandbox.io/s/draggable-slider-forked-dw6ele

**Comment by BoyYangzai** [comment]
@miloszmaki 
hello 
i just adjust the css 
maybe it's normal reder?
can you describe carefully more

**Comment by miloszmaki** [comment]
@BoyYangzai 
this is how it looked before (v10.1.1): https://codesandbox.io/s/draggable-slider-izgbr4
before
and this is how it looks now (v10.2.0): https://codesandbox.io/s/draggable-slider-forked-dw6ele
now

**Comment by BoyYangzai** [comment]
@miloszmaki done~
https://github.com/react-component/slider/issues/912

**Pull request merged**